blaze/expr/collections.py

```diff
@@ -708,8 +710,45 @@ def isin(expr, keys):
     return IsIn(expr, frozenset(keys))
 
 
+class Shift(ElemWise):
+    """ Shift a column backward or forward by N elements
+
+    Parameters
+    ----------
+    expr : Expr
+        The expression to shift. This expression's dshape should be columnar
+    n : int
+        The number of elements to shift by. If n < 0 then shift backward,
+        if n == 0 do nothing, else shift forward.
+    """
+    __slots__ = '_hash', '_child', 'n'
+
+    @property
+    def schema(self):
+        measure = self._child.schema.measure
+
+        # if we are not shifting or we are already an Option type then return
+        # the child's schema
+        if not self.n or isinstance(measure, Option):
```

Comment:
I'm going to leave it as is for now. If this becomes a bottleneck in the future then it's very easy to change